src/Provider/Provider.Web/Orchestrators/VacanciesOrchestrator.cs

```diff
@@ -17,21 +20,39 @@ namespace Esfa.Recruit.Provider.Web.Orchestrators
 {
     public class VacanciesOrchestrator
     {
+        private const int ClosingSoonDays = 5;
         private const int VacanciesPerPage = 25;
-        private readonly IProviderVacancyClient _client;
+        private readonly IProviderVacancyClient _providerVacancyClient;
+        private readonly IRecruitVacancyClient _recruitVacancyClient;
         private readonly ITimeProvider _timeProvider;
-        private const int ClosingSoonDays = 5;
+        private readonly IProviderAlertsViewModelFactory _providerAlertsViewModelFactory;
 
-        public VacanciesOrchestrator(IProviderVacancyClient client, ITimeProvider timeProvider)
+        public VacanciesOrchestrator(
+            IProviderVacancyClient providerVacancyClient,
+            IRecruitVacancyClient recruitVacancyClient,
+            ITimeProvider timeProvider,
+            IProviderAlertsViewModelFactory providerAlertsViewModelFactory)
         {
-            _client = client;
+            _providerVacancyClient = providerVacancyClient;
+            _recruitVacancyClient = recruitVacancyClient;
             _timeProvider = timeProvider;
+            _providerAlertsViewModelFactory = providerAlertsViewModelFactory;
         }
 
         public async Task<VacanciesViewModel> GetVacanciesViewModelAsync(
-            long ukprn, string filter, int page, string searchTerm)
+            VacancyUser user, string filter, int page, string searchTerm)
         {
-            var vacancies = await GetVacanciesAsync(ukprn);
+            var getDashboardTask = _providerVacancyClient.GetDashboardAsync(user.Ukprn.Value, createIfNonExistent: true);
+            var getUserDetailsTask = _recruitVacancyClient.GetUsersDetailsAsync(user.UserId);
+
+            await Task.WhenAll(getDashboardTask, getUserDetailsTask);
+
+            var dashboard = getDashboardTask.Result;
+            var userDetails = getUserDetailsTask.Result;
+
+            var alerts = _providerAlertsViewModelFactory.Create(dashboard, userDetails);
+
+            var vacancies = new List<VacancySummary>(dashboard?.Vacancies ?? Array.Empty<VacancySummary>());
```

Comment:
nice I always used `Enumerable.Empty<Type>().ToArray()` because I didn't know of `Array.Empty<Type>()`